domain/agentprovisioner/state/state.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/canonical/sqlair"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/core/database"
+	"github.com/juju/juju/domain"
+)
+
+// State is responsible for accessing the controller/model DB to retrieve the
+// controller/model config keys required for the container config.
+type State struct {
+	*domain.StateBase
+}
+
+// NewState creates a new State object.
+func NewState(modelFactory database.TxnRunnerFactory) *State {
+	return &State{
+		StateBase: domain.NewStateBase(modelFactory),
+	}
+}
+
+// GetModelConfigKeyValues returns the values of the specified model config
+// keys from the model database.
```

Comment:
Can you update the docs to say what happens if a key isn't found of if no keys are supplied.